src/ebonite/ext/sqlalchemy/repository.py

```diff
@@ -44,19 +44,27 @@ def __init__(self, db_uri: str):
         self._active_session = None
 
     @contextlib.contextmanager
-    def _session(self, commit=True) -> Session:
```

Comment:
actually dont remember why i did it like this, but if tests pass, everything must be okay)

Reply:
In the previous version we committed transaction multiple times in several methods. Now we do it once (at the end) except for the creation case where we need to obtain auto-incrementing `id` before leaving transaction.